nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-settings.js

```diff
@@ -1925,6 +1931,885 @@
+            var icon = '', label = '';
+            if (dataContext.status.validationStatus === 'VALIDATING') {
+                icon = 'validating fa fa-spin fa-circle-notch';
+                label = 'Validating';
+            } else if (dataContext.status.validationStatus === 'INVALID') {
+                icon = 'invalid fa fa-warning';
+                label = 'Invalid';
+            } else {
+                if (dataContext.status.runStatus === 'DISABLED') {
+                    icon = 'disabled icon icon-enable-false"';
+                    label = 'Disabled';
+                } else if (dataContext.status.runStatus === 'ENABLED') {
+                    icon = 'enabled fa fa-flash';
+                    label = 'Enabled';
+                }
+            }
+
+            // format the markup
+            var formattedValue = '<div layout="row"><div class="' + icon + '"></div>';
+            return formattedValue + '<div class="status-text">' + label + '</div></div>';
+        };
+
+        var flowAnalysisRuleActionFormatter = function (row, cell, value, columnDef, dataContext) {
+            var markup = '';
+
+            var canWrite = dataContext.permissions.canWrite;
+            var canRead = dataContext.permissions.canRead;
+            var canOperate = canWrite || (dataContext.operatePermissions && dataContext.operatePermissions.canWrite);
+
+            var isDisabled = dataContext.status.runStatus === 'DISABLED';
+
+            if (canRead) {
+                if (canWrite && isDisabled) {
+                    markup += '<div class="pointer edit-flow-analysis-rule fa fa-gear" title="Configure"></div>';
+                } else {
+                    markup += '<div class="pointer view-flow-analysis-rule fa fa-gear" title="View Configuration"></div>';
+                }
+            }
+
+            if (canOperate) {
+                if (dataContext.status.runStatus === 'ENABLED') {
+                    markup += '<div class="pointer disable-flow-analysis-rule icon icon-enable-false" title="Disable"></div>';
+                } else if (isDisabled && dataContext.status.validationStatus === 'VALID') {
+                    markup += '<div class="pointer enable-flow-analysis-rule fa fa-flash" title="Enable"></div>';
+                }
+            }
+
+            if (isDisabled && canRead && canWrite && dataContext.component.multipleVersionsAvailable === true) {
+                markup += '<div title="Change Version" class="pointer change-version-flow-analysis-rule fa fa-exchange"></div>';
+            }
+
+            if (isDisabled && canRead && canWrite && nfCommon.canModifyController()) {
+                markup += '<div title="Remove" class="pointer delete-flow-analysis-rule fa fa-trash"></div>';
+            }
+
+            if (canRead && canWrite && dataContext.component.persistsState === true) {
+                markup += '<div title="View State" class="pointer view-state-flow-analysis-rule fa fa-tasks"></div>';
```

Comment:
Do Rules persist State?

Reply:
Yes the plan is to allow writing rules that need state.

Reply:
Is there an example of a Rule that is `Stateful`? It would be great to verify the View/Clear State actions work.

Reply:
Not with this change. Only one rule is added, the `DisallowComponentType` which doesn't hold state.